app/api/llm-hint/route.ts

```diff
@@ -1,31 +1,100 @@
 import { NextRequest, NextResponse } from "next/server";
 import { createClient } from "@/utils/supabase/server";
 import { createClient as createServiceClient } from "@supabase/supabase-js";
-import OpenAI from "openai";
+import { AzureChatOpenAI, ChatOpenAI } from "@langchain/openai";
+import { ChatAnthropic } from "@langchain/anthropic";
+import { ChatPromptTemplate } from "@langchain/core/prompts";
+
 import * as Sentry from "@sentry/nextjs";
 import { GraderResultTestExtraData } from "@/utils/supabase/DatabaseTypes";
 
-const openai = new OpenAI({
-  apiKey: process.env.OPENAI_API_KEY
-});
+async function getChatModel({
+  model,
+  provider,
+  temperature,
+  maxTokens,
+  maxRetries,
+  account
+}: {
+  model: string;
+  provider: "openai" | "azure" | "anthropic";
+  temperature?: number;
+  maxTokens?: number;
+  maxRetries?: number;
+  account?: string;
```

Comment:
Is the account just used for testing? i.e., switching between API keys?

Reply:
for co-tenancy on pawtograder, where different classes have different inference API keys